paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/clone/CloneFileInfo.java

```diff
@@ -18,29 +18,35 @@
 
 package org.apache.paimon.flink.clone;
 
+import javax.annotation.Nullable;
+
 /** The information of copy file. */
 public class CloneFileInfo {
-
-    private final String sourceFilePath;
-    private final String filePathExcludeTableRoot;
+    @Nullable private final String sourceFilePath;
+    @Nullable private final String filePathExcludeTableRoot;
     private final String sourceIdentifier;
     private final String targetIdentifier;
+    private final long snapshotId;
```

Comment:
Please remove this field, see my comments for operators.

Reply:
I think we can not remove this filed. because if do not provided the snapshotId. when we do job in CopyManifestFilesOperator, we can not just pick the data files in manifest file, because the data files maybe delete in another mainifest file. 

for example, in snapshot1, we add one data file `data-file1.parquet` in `manifest-file1` ; in snapshot 2, we add one data file `data-file2.parquet` and **delete** `data-file1.parquet` in `manifest-file2`. And these two manifest files were processed in two separate tasks, when processing `manifest-file1` and copy `data-file1.parquet`,  the job will fail.

So we can not just pick the data files in manifest file.  I think we still need the snapshot id. What do you think?

Reply:
I think we don't need to worry about deleting files in the manifest anymore. We just need to see which data files we need. Although we may copy extra files, it won't cause any problems

Reply:
The problem is that, when snapshot1 is expired, then the `data-file1.parquet` will be deleted. when we read the `manifest-file1` and copy  `data-file1.parquet`, it will always fail, because the file do not exist.